File: app/src/main/java/com/gerigorila/agenticmigrationdemo/presentation/home/HomeFragment.java
package com.gerigorila.agenticmigrationdemo.presentation.home;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.gerigorila.agenticmigrationdemo.R;
import com.gerigorila.agenticmigrationdemo.presentation.productlist.ProductListFragment;

public class HomeFragment extends Fragment implements HomeContract.View {

    private HomePresenter presenter;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_home, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        presenter = new HomePresenter(this);

        Button btnStart = view.findViewById(R.id.btn_start);
        btnStart.setOnClickListener(v -> presenter.onStartClicked());
    }

    @Override
    public void navigateToProductList() {
        requireActivity().getSupportFragmentManager()
                .beginTransaction()
                .replace(R.id.fragment_container, new ProductListFragment())
                .addToBackStack(null)
                .commit();
    }
}

File: app/src/main/java/com/gerigorila/agenticmigrationdemo/presentation/home/HomeFragment.kt
package com.gerigorila.agenticmigrationdemo.presentation.home

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import androidx.fragment.app.Fragment
import com.gerigorila.agenticmigrationdemo.R
import com.gerigorila.agenticmigrationdemo.presentation.productlist.ProductListFragment

class HomeFragment : Fragment() {

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View = inflater.inflate(R.layout.fragment_home, container, false)

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        view.findViewById<Button>(R.id.btn_start).setOnClickListener {
            navigateToProductList()
        }
    }

    private fun navigateToProductList() {
        requireActivity().supportFragmentManager
            .beginTransaction()
            .replace(R.id.fragment_container, ProductListFragment())
            .addToBackStack(null)
            .commit()
    }
}
